mlflow/store/artifact/uc_volume_artifact_repo.py

```diff
@@ -0,0 +1,236 @@
+import os
+import posixpath
+from pathlib import Path
+from typing import Optional
+
+import mlflow.utils.databricks_utils
+from mlflow.entities import FileInfo
+from mlflow.environment_variables import MLFLOW_ENABLE_UC_VOLUME_FUSE_ARTIFACT_REPO
+from mlflow.exceptions import MlflowException
+from mlflow.protos.databricks_pb2 import INVALID_PARAMETER_VALUE
+from mlflow.store.artifact.artifact_repo import ArtifactRepository
+from mlflow.store.artifact.local_artifact_repo import LocalArtifactRepository
+from mlflow.utils.databricks_utils import get_databricks_host_creds
+from mlflow.utils.file_utils import relative_path_to_artifact_path
+from mlflow.utils.request_utils import augmented_raise_for_status
+from mlflow.utils.rest_utils import http_request
+from mlflow.utils.uri import (
+    get_databricks_profile_uri_from_artifact_uri,
+    is_databricks_model_registry_artifacts_uri,
+    is_valid_uc_volumes_uri,
+    remove_databricks_profile_info_from_artifact_uri,
+    strip_scheme,
+)
+
+# https://docs.databricks.com/api/workspace/files
+DIRECTORIES_API_ENDPOINT = "/api/2.0/fs/directories"
+FILES_API_ENDPOINT = "/api/2.0/fs/files"
+DOWNLOAD_CHUNK_SIZE = 1024
+
+
+class UCVolumesArtifactRepository(ArtifactRepository):
+    """
+    Stores artifacts on UC Volumes using the Files REST API.
+    """
+
+    def __init__(self, artifact_uri):
+        if not is_valid_uc_volumes_uri(artifact_uri):
+            raise MlflowException(
+                message=(
+                    f"UC volume URI must be of the form "
```

Comment:
```suggestion
                    f"UC Volumes URI must be of the form "
```
Might be more recognizable by the product name :)